### PR TITLE
Added new search

### PR DESCRIPTION
**Pull request by khoroshevskyi**
## Changes
- Added new search: bivec text to bed search

### CODE REVIEW

File: bbconf/config_parser/bedbaseconfig.py [hunk 1]
```diff
@@ -94,14 +98,14 @@ def db_engine(self) -> BaseEngine:
...
-        """
-        return self._t2bsi
+    # @property
+    # def t2bsi(self) -> Union[Text2BEDSearchInterface, None]:
+    #     """
+    #     Get text2bednn object
+    #
+    #     :return: text2bednn object
+    #     """
+    #     return self._t2bsi
```
Comment (donaldcampbelljr) [inline bbconf/config_parser/bedbaseconfig.py:108]:
```suggestion
```

File: bbconf/config_parser/bedbaseconfig.py [hunk 2]
```diff
@@ -205,6 +219,35 @@ def _init_qdrant_backend(self) -> QdrantBackend:
...
 
+    def _init_qdrant_text_backend(self) -> QdrantBackend:
+        """
+        Create qdrant client text embedding object using credentials provided in config file
+
+        :return: QdrantClient
+        """
+
+        return QdrantBackend(
+            dim=384,
```
Comment (donaldcampbelljr) [inline bbconf/config_parser/bedbaseconfig.py:230]: this just looks funny. why is it this specific number?

File: bbconf/config_parser/bedbaseconfig.py [hunk 3]
```diff
@@ -205,6 +219,35 @@ def _init_qdrant_backend(self) -> QdrantBackend:
...
+        Create BiVectorSearchInterface object using credentials provided in config file
+
+        :return: BiVectorSearchInterface
+        """
+
+        search_backend = BiVectorBackend(
+            metadata_backend=self._qdrant_text_engine, bed_backend=self._qdrant_engine
+        )
+        search_interface = BiVectorSearchInterface(
+            backend=search_backend, query2vec="sentence-transformers/all-MiniLM-L6-v2"
```
Comment (donaldcampbelljr) [inline bbconf/config_parser/bedbaseconfig.py:247]: For the future, what is the difference between backend and interface? Is interface the correct term here?

File: bbconf/modules/bedfiles.py [hunk 4]
```diff
@@ -769,7 +768,7 @@ def _embed_file(self, bed_file: Union[str, RegionSet]) -> np.ndarray:
         """
         Create embeding for bed file
 
-        :param bed_id: bed file id
+        :param bed_file: bed file id
```
Comment (donaldcampbelljr) [inline bbconf/modules/bedfiles.py:771]:
```suggestion
        :param bed_file: bed file path or regionset
```

File: bbconf/modules/bedfiles.py [hunk 5]
```diff
@@ -806,9 +805,10 @@ def text_to_bed_search(
         :return: list of bed file metadata
         """
         _LOGGER.info(f"Looking for: {query}")
-        _LOGGER.info(f"Using backend: {self._config.t2bsi}")
+        # _LOGGER.info(f"Using backend: {self._config.t2bsi}")
+        # results = self._config.t2bsi.query_search(query, limit=limit, offset=offset)
```
Comment (donaldcampbelljr) [inline bbconf/modules/bedfiles.py:809]:
```suggestion
```

**Review by donaldcampbelljr — APPROVED** [review]
good to go

File: bbconf/config_parser/bedbaseconfig.py (hunk 3 above)
Comment (khoroshevskyi) [reply bbconf/config_parser/bedbaseconfig.py]: @ClaudeHu

File: bbconf/config_parser/bedbaseconfig.py (hunk 2 above)
Comment (khoroshevskyi) [reply bbconf/config_parser/bedbaseconfig.py:220]: @nleroy917

Comment (nleroy917) [reply bbconf/config_parser/bedbaseconfig.py:220]: Thats the dimensionality of the embeddings for `sentence-transformers/all-MiniLM-L6-v2`. So its fine **iff** the user will never be able to parametrize this particular variable.

Regardless, I agree it should be annotated with a comment or a constant that is-self-descriptive (i.e. `EMBEDDING_DIMENSION` to help explain whats going on)

**Pull request merged**